### PR TITLE
Canary: add random start time for the periodic metric test query

### PR DESCRIPTION
**Pull request by slim-bean**
**What this PR does / why we need it**:

When running a lot of canaries it becomes common that they have a highly overlapping window where they all run the metric test query at the same time, this PR introduces some randomness in the first execution window such that all the canaries should approximately distribute over the size of the interval for which they run the metric test query.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Is this an important fix or new feature? Add an entry in the `CHANGELOG.md`.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

**Review by DylanGuedes — APPROVED** [review]
lgtm

File: pkg/canary/comparator/comparator.go [hunk 1]
```diff
@@ -246,7 +247,10 @@ func (c *Comparator) Size() int {
 
 func (c *Comparator) run() {
 	t := time.NewTicker(c.pruneInterval)
-	mt := time.NewTicker(c.metricTestInterval)
+	// Use a random tick up to the interval for the first tick
+	firstMt := true
+	rand.Seed(time.Now().UnixNano())
+	mt := time.NewTicker(time.Duration(rand.Int63n(c.metricTestInterval.Nanoseconds())))
```
Comment (trevorwhitney) [inline pkg/canary/comparator/comparator.go:253]: I think the default for this is `1hr`, so we delay the first tick between 0-3600s?

**Review by trevorwhitney — APPROVED** [review]
lgtm! thanks ed!

File: pkg/canary/comparator/comparator.go [hunk 2]
```diff
@@ -285,6 +289,11 @@ func (c *Comparator) run() {
 				go c.metricTest(time.Now())
 			}
 			c.metTestMtx.Unlock()
+			if firstMt {
+				// After the first tick which is at a random time, resume at the normal periodic interval
+				firstMt = false
+				mt.Reset(c.metricTestInterval)
+			}
```
Comment (jeschkies) [inline pkg/canary/comparator/comparator.go:296]: I wonder whether a sleep would be easier to understand.

File: pkg/canary/comparator/comparator.go (hunk 1 above)
Comment (slim-bean) [reply pkg/canary/comparator/comparator.go:253]: assuming I did this correctly that's the idea, the base unit of a Duration is nanoseconds so we just spread the start time of this test for every canary randomly over the interval

File: pkg/canary/comparator/comparator.go (hunk 2 above)
Comment (slim-bean) [reply pkg/canary/comparator/comparator.go:296]: a sleep would be more complicated to implement without affecting the other tickers in the select. We don't want to interfere with the spot check test for example, and make it wait up to 1hr before running.

Comment (jeschkies) [reply pkg/canary/comparator/comparator.go:296]: Ok. Then let's go with this solution :slightly_smiling_face:

**Review by jeschkies — APPROVED** [review]

**Pull request merged**